tagstudio/tests/qt/test_tag_widget.py

```diff
@@ -108,3 +108,27 @@ def test_tag_widget_edit(qtbot, qt_driver, library, entry_full):
     assert isinstance(panel, BuildTagPanel)
     assert panel.tag.name == tag.name
     assert panel.name_field.text() == tag.name
+
+def test_tag_widget_autocomplete(qtbot, qt_driver, library):
+    # Given
+    entry = next(library.get_entries(with_joins=True))
+    field = entry.tag_box_fields[0]
+
+    tag_widget = TagBoxWidget(field, "title", qt_driver)
+    tag_widget.driver.selected = [0]
+
+    qtbot.add_widget(tag_widget)
+
+    # Test autocomplete
+    tag_widget.tag_entry.setText("arch")
+    tag_widget.tag_entry.returnPressed.emit()
+
+    entry = next(library.get_entries(with_joins=True)) # Update entry
+    assert len(entry.tags) == 2
+
+    # Test unmatched autocomplete
+    tag_widget.tag_completer.activated.emit("missing")
+
+    entry = next(library.get_entries(with_joins=True)) # Update entry
+    assert len(entry.tags) == 2
```

Comment:
I'm not sure I understand the check here. Is it that searching for `arch` adds new tag to the Entry (thus len(tags) == 2), but then searching for `missing` afterwards doesnt yield any results, so the `len(tags) == 2` is still valid? 
If so, could you add a check `len(tags) == 1` before adding the first tag to confirm there is only one tag before adding `arch`? thanks.

Reply:
Yep, that's exactly it. I will add the extra check today.